### PR TITLE
Map username to email of the user

### PR DESCRIPTION
**Pull request by jinzeee**
## Description of what I changed

Map email username to the notification address if email as the username is enabled and add test to map email username to the notification address if email as username flag is enabled

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-3650

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by ODORA0** [comment]
Did you look at this https://wiki.openmrs.org/display/docs/Pull+Request+Tips? There is need for change in that commit message

File: api/src/test/java/org/openmrs/api/UserServiceTest.java [hunk 1]
```diff
@@ -32,6 +33,7 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
+import org.openmrs.GlobalProperty;
```
Comment (Am-Coder) [inline api/src/test/java/org/openmrs/api/UserServiceTest.java]: Where are you using this?

Comment (joachim-junior) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java]: Please come out your code @jinzeee so we don't stress a lot in finding stack traces.

Comment (joachim-junior) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java]: Sorry, meant comment out**

Comment (joachim-junior) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java]: Also i think you should go through the documentation on how to submit a pull request again. You can find it [here ]( https://wiki.openmrs.org/display/docs/Pull+Request+Tips)

**Pull request closed without merging**